src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DnsEndPointWithProperties.cs

```diff
@@ -9,18 +9,18 @@ namespace System.Net.Http
     // Passed to a connection factory, merges allocations for the DnsEndPoint and connection properties.
     internal sealed class DnsEndPointWithProperties : DnsEndPoint, IConnectionProperties
     {
-        public HttpRequestMessage InitialRequest { get; }
+        private readonly HttpRequestMessage _initialRequest;
 
         public DnsEndPointWithProperties(string host, int port, HttpRequestMessage initialRequest) : base(host, port)
         {
-            InitialRequest = initialRequest;
+            _initialRequest = initialRequest;
```

Comment:
I don't quite understand why this class exists. 

I get that we want to pass the HttpRequest through as a property, but why aren't we just doing that directly via SocketsConnectionFactory? Why do we subclass DnsEndPoint?

@scalablecory Thoughts?

Reply:
It's the property provider (`IConnectionProperties` implementation) that is aware of `HttpRequestMessage`. `SocketsConnectionFactory` does not know about HTTP stuff.

We subclass `DnsEndPoint` to reduce allocations. (A DnsEndpoint instance has to be created anyways.)

Reply:
Ah, okay, I understand now. There's a comment above but it wasn't entirely clear to me.